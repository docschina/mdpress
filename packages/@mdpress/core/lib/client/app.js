/* global MDPRESS_TEMP_PATH */
import React,{ useEffect } from 'react';
import { Switch,Redirect,Route } from 'react-router-dom';
import { ScrollContext } from 'react-router-scroll-4';
import store from '@app/store';
import { siteData } from '@internal/siteData';
import appEnhancers from '@internal/app-enhancers';
import globalUIComponents from '@internal/global-ui';
import { routes } from '@internal/routes';
import { handleRedirectForCleanUrls } from './redirect.js';
import sandbox from '@internal/sandbox';
import mdpress from './store';
import { getElementPosition, normalizeOffset,getEnv } from './util';

const isServer = getEnv() === 'server';
const mdConfig = mdpress.mdConfig;
// suggest dev server restart on base change
if (module.hot) {
  const prevBase = siteData.base;
  module.hot.accept(MDPRESS_TEMP_PATH + '/internal/siteData.js', () => {
    if (siteData.base !== prevBase) {
      window.alert(
        '[mdpress] Site base has changed. '
                + 'Please restart dev server to ensure correct asset paths.'
      );
    }
  });
}

const shouldUpdateScroll = (prevRouterProps, { location }) => {
  if (location.hash) {
    if (store.$get('disableScrollBehavior')) {
      return false;
    }

    const dom = document.querySelector(`${decodeURIComponent(location.hash)}`);
    if (dom){
      const position = getElementPosition(dom,normalizeOffset(dom));
      window.scrollTo(position.x, position.y);
    }
  } else {
    return [0,0];
  }
};

function App(props) {
  const { shouldUpdateScroll } = props;
  useEffect(() => {
    setTimeout(()=>{
      shouldUpdateScroll && shouldUpdateScroll(null, { location: window.location });
    },1000);
  },[]);

  const { routes,hooks } = props;

  hooks.forEach(hook => {
    typeof hook === 'function' && hook();
  });
  handleRedirectForCleanUrls(routes);

  const children = <React.Fragment>
    <Switch>
      {routes.map(route =>
        route.redirect ? <Redirect key={route.path} strict={false} exact from={route.path} to={route.redirect}/> : <Route exact strict={false} path={route.path} key={route.path} component={route.component} render={route.render}/>
      )}
    </Switch>
    {globalUIComponents.map((name,index) => {
      const Comp = sandbox[name] || (() => null);
      return <Comp key={index}/>;
    })}
  </React.Fragment>;


  return isServer ? children : <ScrollContext shouldUpdateScroll={shouldUpdateScroll}>
    {children}
  </ScrollContext>;
}

export async function createApp (isServer) {
  const routerBase = typeof window !== 'undefined' && window.__MDPRESS_ROUTER_BASE__
    ? window.__MDPRESS_ROUTER_BASE__
    : (siteData.routerBase || siteData.base);

  let hooks = [];
  routes.shouldUpdateScroll = shouldUpdateScroll;
  try {
    await Promise.all(
      appEnhancers
        .filter(enhancer => typeof enhancer === 'function')
        .map(enhancer => enhancer({ sandbox, routes, siteData, isServer,mdConfig,hooks }))
    );
  } catch (e) {
    console.error(e);
  }

  return {
    app: <App
      routes={routes}
      siteData={siteData}
      hooks={hooks}
      shouldUpdateScroll={routes.shouldUpdateScroll}
    />,
    routerBase
  };
}
