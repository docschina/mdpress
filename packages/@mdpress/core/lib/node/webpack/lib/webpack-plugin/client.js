'use strict';
var __importDefault = (this && this.__importDefault) || function (mod) {
  return (mod && mod.__esModule) ? mod : { 'default': mod };
};
Object.defineProperty(exports, '__esModule', { value: true });
var util_1 = require('./util');
var hash_sum_1 = __importDefault(require('hash-sum'));
var lodash_1 = require('lodash');
var ReactSSRClientPlugin = /** @class */ (function () {
  function ReactSSRClientPlugin(options) {
    if (options === void 0) { options = {}; }
    this.options = Object.assign({
      filename: 'react-ssr-client-manifest.json',
    }, options);
  }
  ReactSSRClientPlugin.prototype.apply = function (compiler) {
    var _this = this;
    util_1.onEmit(compiler, 'react-client-plugin', function (compilation, cb) {
      var stats = compilation.getStats().toJson();
      var allFiles = lodash_1.uniq(stats.assets.map(function (a) { return a.name; }));
      var initialFiles = lodash_1.uniq(Object.keys(stats.entrypoints)
        .map(function (name) { return stats.entrypoints[name].assets; })
        .reduce(function (assets, all) { return all.concat(assets); }, [])
        .filter(function (file) { return util_1.isJS(file) || util_1.isCSS(file); }));
      var asyncFiles = allFiles
        .filter(function (file) { return util_1.isJS(file) || util_1.isCSS(file); })
        .filter(function (file) { return initialFiles.indexOf(file) < 0; });
      var manifest = {
        publicPath: stats.publicPath,
        all: allFiles,
        initial: initialFiles,
        async: asyncFiles,
        modules: {
          /* [identifier: string]: Array<index: number> */
        },
      };
      var assetModules = stats.modules.filter(function (m) { return m.assets.length; });
      var fileToIndex = function (file) { return manifest.all.indexOf(file); };
      stats.modules.forEach(function (m) {
        // ignore modules duplicated in multiple chunks
        if (m.chunks.length === 1) {
          var cid_1 = m.chunks[0];
          var chunk = stats.chunks.find(function (c) { return c.id === cid_1; });
          if (!chunk || !chunk.files) {
            return;
          }
          var id = m.identifier.replace(/\s\w+$/, ''); // remove appended hash
          var files_1 = (manifest.modules[hash_sum_1.default(id)] = chunk.files.map(fileToIndex));
          // find all asset modules associated with the same chunk
          assetModules.forEach(function (module) {
            if (module.chunks.some(function (chunkId) { return chunkId === cid_1; })) {
              files_1.push.apply(files_1, module.assets.map(fileToIndex));
            }
          });
        }
      });
      // const debug = (file, obj) => {
      //   require('fs').writeFileSync(__dirname + '/' + file, JSON.stringify(obj, null, 2))
      // }
      // debug('stats.json', stats)
      // debug('client-manifest.json', manifest)
      var json = JSON.stringify(manifest, null, 2);
      compilation.assets[_this.options.filename] = {
        source: function () { return json; },
        size: function () { return json.length; },
      };
      cb();
    });
  };
  return ReactSSRClientPlugin;
}());
exports.ReactSSRClientPlugin = ReactSSRClientPlugin;
//# sourceMappingURL=client.js.map