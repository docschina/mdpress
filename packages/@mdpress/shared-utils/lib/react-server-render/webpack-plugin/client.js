"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReactSSRClientPlugin = void 0;
const util_1 = require("./util");
const hash_sum_1 = __importDefault(require("hash-sum"));
const lodash_1 = require("lodash");
const styleJsChunkreg = /styles\.\w{8}\.js$/;
class ReactSSRClientPlugin {
    constructor(options = {}) {
        this.options = Object.assign({
            filename: 'react-ssr-client-manifest.json',
        }, options);
    }
    apply(compiler) {
        util_1.onEmit(compiler, 'react-client-plugin', (compilation, cb) => {
            const stats = compilation.getStats().toJson();
            const allFiles = lodash_1.uniq(stats.assets
                .map(a => a.name)
                // Avoid preloading / injecting the style chunk
                .filter(file => !styleJsChunkreg.test(file)));
            const initialFiles = lodash_1.uniq(Object.keys(stats.entrypoints)
                .map(name => stats.entrypoints[name].assets)
                .reduce((assets, all) => all.concat(assets), [])
                .filter(file => util_1.isJS(file) || util_1.isCSS(file))
                .filter(function (file) { return !styleJsChunkreg.test(file); }));
            const asyncFiles = allFiles
                .filter(file => util_1.isJS(file) || util_1.isCSS(file))
                .filter(file => initialFiles.indexOf(file) < 0);
            const manifest = {
                publicPath: stats.publicPath,
                all: allFiles,
                initial: initialFiles,
                async: asyncFiles,
                modules: {
                /* [identifier: string]: Array<index: number> */
                },
            };
            const assetModules = stats.modules.filter(m => m.assets.length);
            const fileToIndex = file => manifest.all.indexOf(file);
            stats.modules.forEach(m => {
                // ignore modules duplicated in multiple chunks
                if (m.chunks.length === 1) {
                    const cid = m.chunks[0];
                    const chunk = stats.chunks.find(c => c.id === cid);
                    if (!chunk || !chunk.files) {
                        return;
                    }
                    const id = m.identifier.replace(/\s\w+$/, ''); // remove appended hash
                    const files = (manifest.modules[hash_sum_1.default(id)] = chunk.files.map(fileToIndex));
                    // find all asset modules associated with the same chunk
                    assetModules.forEach(module => {
                        if (module.chunks.some(chunkId => chunkId === cid)) {
                            files.push(...module.assets.map(fileToIndex));
                        }
                    });
                }
            });
            // const debug = (file, obj) => {
            //   require('fs').writeFileSync(__dirname + '/' + file, JSON.stringify(obj, null, 2))
            // }
            // debug('stats.json', stats)
            // debug('client-manifest.json', manifest)
            const json = JSON.stringify(manifest, null, 2);
            if (this.options && this.options.filename) {
                compilation.assets[this.options.filename] = {
                    source: () => json,
                    size: () => json.length,
                };
            }
            cb();
        });
    }
}
exports.ReactSSRClientPlugin = ReactSSRClientPlugin;
