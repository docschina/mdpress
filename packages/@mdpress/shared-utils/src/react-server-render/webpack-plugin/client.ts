import { isCSS, isJS, onEmit } from './util';

import hash from 'hash-sum';
import { uniq } from 'lodash';

const styleJsChunkreg = /styles\.\w{8}\.js$/;

export class ReactSSRClientPlugin {
    options: {
        filename?: string;
    }

    constructor(options = {}) {
      this.options = Object.assign(
        {
          filename: 'react-ssr-client-manifest.json',
        },
        options,
      );
    }

    apply(compiler) {
      onEmit(compiler, 'react-client-plugin', (compilation, cb) => {
        const stats = compilation.getStats().toJson();

        const allFiles = uniq<string>(stats.assets
          .map(a => a.name)
        // Avoid preloading / injecting the style chunk
          .filter(file => !styleJsChunkreg.test(file))
        );

        const initialFiles = uniq(
          Object.keys(stats.entrypoints)
            .map(name => stats.entrypoints[name].assets)
            .reduce((assets, all) => all.concat(assets), [])
            .filter(file => isJS(file) || isCSS(file))
            .filter(function (file) { return !styleJsChunkreg.test(file); })

        );

        const asyncFiles = allFiles
          .filter(file => isJS(file) || isCSS(file))
          .filter(file => initialFiles.indexOf(file) < 0);

        const manifest = {
          publicPath: stats.publicPath,
          all: allFiles,
          initial: initialFiles,
          async: asyncFiles,
          modules: {
            /* [identifier: string]: Array<index: number> */
          },
        };

        const assetModules = stats.modules.filter(m => m.assets.length);
        const fileToIndex = file => manifest.all.indexOf(file);
        stats.modules.forEach(m => {
          // ignore modules duplicated in multiple chunks
          if (m.chunks.length === 1) {
            const cid = m.chunks[0];
            const chunk = stats.chunks.find(c => c.id === cid);
            if (!chunk || !chunk.files) {
              return;
            }
            const id = m.identifier.replace(/\s\w+$/, ''); // remove appended hash
            const files = (manifest.modules[hash(id)] = chunk.files.map(
              fileToIndex,
            ));
            // find all asset modules associated with the same chunk
            assetModules.forEach(module => {
              if (module.chunks.some(chunkId => chunkId === cid)) {
                files.push(...module.assets.map(fileToIndex));
              }
            });
          }
        });

        // const debug = (file, obj) => {
        //   require('fs').writeFileSync(__dirname + '/' + file, JSON.stringify(obj, null, 2))
        // }
        // debug('stats.json', stats)
        // debug('client-manifest.json', manifest)

        const json = JSON.stringify(manifest, null, 2);
        if (this.options && this.options.filename) {
          compilation.assets[this.options.filename] = {
            source: () => json,
            size: () => json.length,
          };
        }
        cb();
      });
    }
}
