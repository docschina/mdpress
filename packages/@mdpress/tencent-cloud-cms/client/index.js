import React from 'react';
import Layout from './Layout';

const isServer = typeof window === 'undefined';
export default ({ routes,siteData }) => {
  const notFoundRoute = routes.find(route => route.path === '\*');
  const NotFoundComp = notFoundRoute.component;

  notFoundRoute.component = function Comp(props) {
    if (isServer) return null;
    return <NotFoundComp {...props}/>;
  };

  routes.unshift({
    name: 'docs',
    path: `${siteData.base}docs/:path`,
    component: Layout
  });
};
