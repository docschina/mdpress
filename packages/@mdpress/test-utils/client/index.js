import Router from './Router';

export {
  Router
};
