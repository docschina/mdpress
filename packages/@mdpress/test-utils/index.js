const getFragment = require('./lib/getFragment');
const getFragments = require('./lib/getFragments');
const createJestRunner = require('./lib/createJestRunner');
const createJestConfig = require('./lib/createJestConfig');

module.exports = {
  getFragment,
  getFragments,
  createJestRunner,
  createJestConfig
};
