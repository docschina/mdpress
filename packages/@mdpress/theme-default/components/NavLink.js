import React from 'react';
import useData from '@app/hooks/data';
import { NavLink as Link } from 'react-router-dom';
import { isExternal, isMailto, isTel, ensureExt } from '../util';
import OutboundLink from '@app/components/OutboundLink';

export default function NavLink(props) {
  const { item = {},className,onFocusout } = props;
  const { $site } = useData();

  const link = ensureExt(item.link);

  const exact = function () {
    if ($site.locales) {
      return Object.keys($site.locales).some(rootLink => rootLink === link);
    }
    return link === '/';
  }();

  const isNonHttpURI = isMailto(link) || isTel(link);

  const target = (() => {
    if (isNonHttpURI) {
      return null;
    }
    if (item.target) {
      return item.target;
    }
    return isExternal(link) ? '_blank' : '';
  })();

  const isBlankTarget = target === '_blank';
  const isInternal = (() => {
    return !isExternal(link) && !isBlankTarget;
  })();

  const rel = (() => {
    if (isNonHttpURI) {
      return null;
    }
    if (item.rel) {
      return item.rel;
    }
    return isBlankTarget ? 'noopener noreferrer' : '';
  })();

  const focusoutAction = () => {
    onFocusout && onFocusout();
  };

  return <React.Fragment>
    {isInternal ?
      <Link
        activeClassName={'router-link-active'}
        exact={exact}
        className={'nav-link ' + className ? className : ''}
        to={link}
        onBlur={focusoutAction}
      >
        {item.text}
      </Link> : <a
        href={link}
        className={'nav-link external'}
        target={target}
        rel={rel}
        onBlur={focusoutAction}
      >
        {item.text}
        {isBlankTarget && <OutboundLink/>}
      </a>}

  </React.Fragment>;
}