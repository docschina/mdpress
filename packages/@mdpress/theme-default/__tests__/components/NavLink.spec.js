import React from 'react';
import { render } from 'enzyme';
import NavLink from '../../components/NavLink';
import { Router } from '@mdpress/test-utils/client';

describe('NavLink', () => {
  test('renders nav link with internal link', () => {
    const item = {
      link: '/',
      text: 'mdpress'
    };

    const wrapper = render(
      <Router><NavLink item={item}></NavLink></Router>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('renders nav link with external link', () => {
    const item = {
      link: 'https://react.docschina.org/',
      text: 'React'
    };
    const wrapper = render(
      <Router><NavLink item={item}></NavLink></Router>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
