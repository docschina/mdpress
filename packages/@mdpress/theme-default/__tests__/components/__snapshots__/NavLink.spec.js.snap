// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink renders nav link with external link 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "nav-link external",
      "href": "https://react.docschina.org/",
      "rel": "noopener noreferrer",
      "target": "_blank",
    },
    "children": Array [
      Object {
        "data": "React",
        "next": Object {
          "attribs": Object {
            "aria-hidden": "true",
            "class": "icon outbound",
            "height": "15",
            "viewBox": "0 0 100 100",
            "width": "15",
            "x": "0px",
            "xmlns": "http://www.w3.org/2000/svg",
            "y": "0px",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "d": "M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z",
                "fill": "currentColor",
              },
              "children": Array [],
              "name": "path",
              "namespace": "http://www.w3.org/2000/svg",
              "next": Object {
                "attribs": Object {
                  "fill": "currentColor",
                  "points": "45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9",
                },
                "children": Array [],
                "name": "polygon",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "fill": undefined,
                  "points": undefined,
                },
                "x-attribsPrefix": Object {
                  "fill": undefined,
                  "points": undefined,
                },
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "d": undefined,
                "fill": undefined,
              },
              "x-attribsPrefix": Object {
                "d": undefined,
                "fill": undefined,
              },
            },
            Object {
              "attribs": Object {
                "fill": "currentColor",
                "points": "45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9",
              },
              "children": Array [],
              "name": "polygon",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "d": "M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z",
                  "fill": "currentColor",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                  "fill": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                  "fill": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "fill": undefined,
                "points": undefined,
              },
              "x-attribsPrefix": Object {
                "fill": undefined,
                "points": undefined,
              },
            },
          ],
          "name": "svg",
          "namespace": "http://www.w3.org/2000/svg",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "aria-hidden": undefined,
            "class": undefined,
            "height": undefined,
            "viewBox": undefined,
            "width": undefined,
            "x": undefined,
            "xmlns": "http://www.w3.org/2000/xmlns/",
            "y": undefined,
          },
          "x-attribsPrefix": Object {
            "aria-hidden": undefined,
            "class": undefined,
            "height": undefined,
            "viewBox": undefined,
            "width": undefined,
            "x": undefined,
            "xmlns": "",
            "y": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
      Object {
        "attribs": Object {
          "aria-hidden": "true",
          "class": "icon outbound",
          "height": "15",
          "viewBox": "0 0 100 100",
          "width": "15",
          "x": "0px",
          "xmlns": "http://www.w3.org/2000/svg",
          "y": "0px",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "d": "M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z",
              "fill": "currentColor",
            },
            "children": Array [],
            "name": "path",
            "namespace": "http://www.w3.org/2000/svg",
            "next": Object {
              "attribs": Object {
                "fill": "currentColor",
                "points": "45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9",
              },
              "children": Array [],
              "name": "polygon",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "fill": undefined,
                "points": undefined,
              },
              "x-attribsPrefix": Object {
                "fill": undefined,
                "points": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "d": undefined,
              "fill": undefined,
            },
            "x-attribsPrefix": Object {
              "d": undefined,
              "fill": undefined,
            },
          },
          Object {
            "attribs": Object {
              "fill": "currentColor",
              "points": "45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9",
            },
            "children": Array [],
            "name": "polygon",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "d": "M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z",
                "fill": "currentColor",
              },
              "children": Array [],
              "name": "path",
              "namespace": "http://www.w3.org/2000/svg",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "d": undefined,
                "fill": undefined,
              },
              "x-attribsPrefix": Object {
                "d": undefined,
                "fill": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "fill": undefined,
              "points": undefined,
            },
            "x-attribsPrefix": Object {
              "fill": undefined,
              "points": undefined,
            },
          },
        ],
        "name": "svg",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "data": "React",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "aria-hidden": undefined,
          "class": undefined,
          "height": undefined,
          "viewBox": undefined,
          "width": undefined,
          "x": undefined,
          "xmlns": "http://www.w3.org/2000/xmlns/",
          "y": undefined,
        },
        "x-attribsPrefix": Object {
          "aria-hidden": undefined,
          "class": undefined,
          "height": undefined,
          "viewBox": undefined,
          "width": undefined,
          "x": undefined,
          "xmlns": "",
          "y": undefined,
        },
      },
    ],
    "name": "a",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "href": undefined,
      "rel": undefined,
      "target": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "href": undefined,
      "rel": undefined,
      "target": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`NavLink renders nav link with internal link 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "aria-current": "page",
      "class": "router-link-active",
      "href": "/",
    },
    "children": Array [
      Object {
        "data": "mdpress",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "a",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "aria-current": undefined,
      "class": undefined,
      "href": undefined,
    },
    "x-attribsPrefix": Object {
      "aria-current": undefined,
      "class": undefined,
      "href": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
